### DISCUSSION TITLE
Number and Rectangle in Animated Dial EX

### DISCUSSION THREAD

**Discussion by alikhan968** [Bodmer/TFT_eSPI#3121] [Q&A - General]
Hi,
I am trying to use Animated dial example on a 800x480 TFT screen.
I was able to move JPG to desire location along with needle and circle but i couldn't find a line that also move rectangle and number into desire position of the display.
Can someone help me with it plz?
Thanks
@Bodmer

**Comment by Bodmer** [comment]
The position of the middle of the meter in the example is 120,120. I suspect thse lines need to be changed:
https://github.com/Bodmer/TFT_eSPI/blob/master/examples/Sprite/Animated_dial/Animated_dial.ino#L87
https://github.com/Bodmer/TFT_eSPI/blob/master/examples/Sprite/Animated_dial/Animated_dial.ino#L208


Replace 120,120 with DIAL_CENTRE_X ,DIAL_CENTRE_Y and adjust lines [26 and 27](#define DIAL_CENTRE_X 120
#define DIAL_CENTRE_Y 120) according to the new position.

**Comment by alikhan968** [comment]
Thanks for your reply @Bodmer
I did these modifications and attached the picture of the results.
**1: Change 120 to 320**
`#define DIAL_CENTRE_X 320
#define DIAL_CENTRE_Y 320`
**2: Change origin from (0, 0) to (160,160)**
`  TJpgDec.drawJpg(DIAL_CENTRE_X/2, DIAL_CENTRE_Y/2, dial, sizeof(dial));`
**3: Replaced (120,120) to (320,320) as you recommended.**
`  bg_color = tft.readPixel(DIAL_CENTRE_X, DIAL_CENTRE_Y); // Get colour from dial centre`

Now i was able to change JPG, needle and small circle position as desire but the number changing rectangle still stay at original position. I tried to change L207 and L208 but it didn't help me
Let me know what else i can change in the code so that numbers rectangle can relocate.
Thanks
![IMG_20240121_153344](https://github.com/Bodmer/TFT_eSPI/assets/81488667/ff2a61fc-3b1f-4675-9953-e931002feaed)

![IMG_20240121_153341](https://github.com/Bodmer/TFT_eSPI/assets/81488667/619a8e61-57c3-424d-824a-495e6cafca36)
